protocols/satisfactory.js

```diff
@@ -0,0 +1,92 @@
+import Core from './core.js'
+
+/**
+ * Satisfactory servers run by default self-signed certificates. This is used to ignore the validation for the requests.
+ */
+const https = {
+  rejectUnauthorized: false
+}
```

Comment:
I'd have to look again through the docs, but if you can provide trusted certificaties in a server we wouldn't want this to be hardcoded, instead, set this value only if it is not provided (for example as in `if (!this.options.port) this.options.port = 27016`).

Reply:
I'm a bit stuck on this one - there is an option in the Satisfactory API where you can provide your own certificate for the REST API. This is documented in the Satisfactory API docs here:
https://satisfactory.wiki.gg/wiki/Dedicated_servers/HTTPS_API#Certificate_Validation_and_Encryption

node-gamedig uses the Got HTTP library under the surface for making HTTP requests. This is limited in that there is the rejectUnauthorized flag which will allow self-signed or expired certificates. There is the checkServerIdentity function, but this function is not available for self-signed certificates (tested extensively and it doesn't run).

https://github.com/sindresorhus/got/blob/main/documentation/5-https.md

I think there are couple of different options:

1. We remove the rejectUnauthorized flag and document node-gamedig doesn't support self-signed certificates for Satisfactory and will only connect if the user has defined a valid certificate for the Satisfactory API.

2. We continue to set the rejectUnauthorized flag to false noting and document the use of self-signed certificates are a security risk. 

3. We implement an socket listener or similar to check the certificate as the connection is open (by the time the protocol file is dealing with it the connection is closed and certificate information is not available). If we do this we could perform some checks around certificate validity (i.e. certificate is not expired, name matches expected values). This might require more libraries to achieve and involved messing with the core.js code which I'm less comfortable around.

Reply:
1: I wouldn't go for this one as I think many users will not setup proper certificates.
3: This one neither, it involves changing quite a lot of structure and I don't quite see the benefits just yet.
2: We could do this considering the fact that many other (HTTP) protocols aren't secured, but we can still go with an `if (!this.options.rejectUnauthorized) this.options.rejectUnauthorized = false` (and add a note about security and this option to disable this behaviour) to make sure the users have control over this.

Reply:
Have gone with Option #3 and documented. I thought about whether to add this to the core.js request but instead chose to set it at the protocol level. Reasoning being is future protocols may wish to use other flags in the https options so better to set at the lower level.